nibabel/wrapstruct.py

```diff
@@ -536,8 +536,18 @@ def __str__(self):
 
         def _getter(obj, key):
             try:
-                return obj.get_value_label(key)
+                val = obj.get_value_label(key)
             except ValueError:
-                return obj[key]
+                val = obj[key]
+            # Convert a dtype('S1') numpy string to a Python str, if possible
+            if isinstance(val, np.ndarray) and val.dtype.type == np.bytes_:
+                try:
+                    tmp = val.tolist().decode()
+                    # Do not return unicode objects (Python 2)
+                    if isinstance(tmp, str):
```

Comment:
Sorry to say - but if it's just for the doctest, I think we should just skip the doctest or use the somewhat-ugly ellipsis to fix the output.  If anything I'd prefer to see the 'b' in front of foo in Python 3 - it makes it clearer what it is.